### DISCUSSION TITLE
How to bundle all styles to single file with same name as entriypoint?

### DISCUSSION THREAD

**Discussion by krutoo** [web-infra-dev/rspack#5774] [Q&A]
According to this guide:
https://webpack.js.org/plugins/mini-css-extract-plugin/#extracting-all-css-in-a-single-file

It is possible to bundle all styles to single file in webpack and mini-css-extract-plugin.

How can i make it in Rspack?

i think `type: "css/mini-extract"` is not suitable for Rspack because there is no mini-css-extract-plugin in my config but there is a native css support

**Comment by JSerFeng** [comment]
You can do it for type: 'css', also set name to cacheGroup

**Comment by BazukenDev** [comment]
I have same problem. Please write some example.

**Comment by JSerFeng** [comment]
Now you can do the same with rspack.cssExtractPlugin, the same configuration with webpack

**Comment by BazukenDev** [comment]
@JSerFeng
So i cant understand what i doing wrong. 

https://github.com/BazukenDev/rspack-scss.git
node v22

File:app/src/js/popup.js 

```
import 'tippy.js/dist/tippy.css';
import 'tippy.js/animations/scale.css';
import 'tippy.js/animations/perspective.css';
import 'tippy.js/dist/svg-arrow.css';
```
This always create a second bundle. Or i just get this error:
ERROR in × Conflict: Multiple assets emit different content to the same filename app-main.css

**Comment by BazukenDev** [comment]
Fix: use **test: /\.s?css$/,**
        
        
        {
          test: /\.s?css$/,
          type: 'javascript/auto',
          parser: {
            'css/auto': {
              namedExports: false,
            },
          },
          use: [
            {
              loader: rspack.CssExtractRspackPlugin.loader,
              options: {
                sourceMap: true,
              }
            },
            {
              loader: 'css-loader',
              options: {
                url: false,
                // import: false,
                // modules: false,
                sourceMap: true,
              },
            },
            {
              loader: 'builtin:lightningcss-loader',
              /** @type {import('@rspack/core').LightningcssLoaderOptions} */
              options: {
                targets: "last 2 years",
                // sourceMap: true
              },
            },
            {
              loader: 'sass-loader',
              options: {
                api: 'modern-compiler',
                implementation: sassEmbedded,
                sourceMap: true,
              },
            },
          ],
        },
